packages/app-mobile/components/DialogManager/hooks/useDialogControl.ts

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Alert, Platform } from 'react-native';
+import { DialogControl, DialogType, MenuChoice, PromptButton, PromptDialogData, PromptOptions } from '../types';
+import { _ } from '@joplin/lib/locale';
+import { useMemo, useRef } from 'react';
+
+type SetPromptDialogs = React.Dispatch<React.SetStateAction<PromptDialogData[]>>;
+
+const useDialogControl = (setPromptDialogs: SetPromptDialogs) => {
+	const nextDialogIdRef = useRef(0);
+
+	const dialogControl: DialogControl = useMemo(() => {
+		const onDismiss = (dialog: PromptDialogData) => {
+			setPromptDialogs(dialogs => dialogs.filter(d => d !== dialog));
+		};
+
+		const defaultButtons = [{ text: _('OK') }];
+		const control: DialogControl = {
+			info: (message: string) => {
+				return new Promise<void>((resolve) => {
+					control.prompt(_('Info'), message, [{
+						text: _('OK'),
+						onPress: () => resolve(),
+					}]);
+				});
+			},
+			error: (message: string) => {
+				return new Promise<void>((resolve) => {
+					control.prompt(_('Error'), message, [{
+						text: _('OK'),
+						onPress: () => resolve(),
+					}]);
+				});
+			},
+			prompt: (title: string, message: string, buttons: PromptButton[] = defaultButtons, options?: PromptOptions) => {
```

Comment:
**Note**: This function was originally in [`DialogManager.tsx`](https://github.com/laurent22/joplin/blob/8e3c81717e24d7dab85bc3f0b34d99839a83ef34/packages/app-mobile/components/DialogManager.tsx#L57).